client/pages/Dashboard/index.jsx

```diff
@@ -213,10 +213,9 @@ class Dashboard extends Component {
           />
           <Route path="/invite/:inviteHash" component={Invite} />
           <Route path="/events" render={() => <EventsPage />} />
-          <Route path="/dashboard" render={() => <EventsPage />} />
+          <Route path="/dashboard" render={() => <EventsPage createEventBtn={() => this.renderCreateEventButton(categories)} />} />
```

Comment:
Great approach on this fix @uchemukolo. It's a pity that we have to deal with this code smell in the client architecture. But to help address the architectural issues we currently have and not add extra complexity to it, I would suggest that we try to move the `renderCreateEventButton` method into the `EventPage` component instead of trying to inject it into it. This will ensure that we are properly encapsulating rendering logic. Passing the button node as a prop here has no contextual meaning and may be more confusing for a new dev why a non-reusable button (by design) is separated from the `container` component that uses it.

Reply:
Your concern is noted. I will work on it.

Reply:
Feedback implemented